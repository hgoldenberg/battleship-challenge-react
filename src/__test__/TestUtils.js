/* eslint-disable react/prop-types */
/* eslint-disable max-len */
import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render as rlRender } from '@testing-library/react';
import aplicationReducer from '../redux/reducers/index';

const store = createStore(aplicationReducer);

export const ProviderMock = ({ children }) => (
  <Provider store={store}>
    {children}
  </Provider>
);

const Wrapper = ({ children }) => <ProviderMock>{children}</ProviderMock>;

export const renderWithRedux = (component, renderOptions) => rlRender(component, { wrapper: Wrapper, ...renderOptions });