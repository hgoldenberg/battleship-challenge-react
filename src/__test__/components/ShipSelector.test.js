/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { renderWithRedux } from '../TestUtils';
import ShipSelector from '../../components/shipSelector/ShipSelector';

const props = {
  shipSaved: true,
  carriers: 0,
  cruisers: 0,
  submarines: 0,
  selectShip: () => {},
  selectOrientation: () => {},
  restartSavedPlayerShip: () => {},
};

describe('<ShipSelector/>', () => {
  test('ShipSelector component render', () => {
    renderWithRedux(<ShipSelector {...props} />);
  });
});

describe('ShipSelector snapshot', () => {
  test('Check the ShipSelector Snapshot', () => {
    const { container } = renderWithRedux(<ShipSelector {...props} />);
    expect(container).toMatchSnapshot();
  });
});