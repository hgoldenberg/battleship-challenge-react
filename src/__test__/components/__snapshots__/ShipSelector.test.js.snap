// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipSelector snapshot Check the ShipSelector Snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm jJALcW"
  >
    <div
      class="sc-AxirZ dWXyjM"
    >
      <div
        class="sc-AxiKw jhuQxN"
      >
        <label>
          Carriers
          :   
          0
        </label>
        <button
          class="sc-AxhUy clkaBI"
        >
          Select
        </button>
      </div>
    </div>
    <div
      class="sc-AxirZ dWXyjM"
    >
      <div
        class="sc-AxiKw jhuQxN"
      >
        <label>
          Cruisers
          :   
          0
        </label>
        <button
          class="sc-AxhUy clkaBI"
        >
          Select
        </button>
      </div>
    </div>
    <div
      class="sc-AxirZ dWXyjM"
    >
      <div
        class="sc-AxiKw jhuQxN"
      >
        <label>
          Submarines
          :   
          0
        </label>
        <button
          class="sc-AxhUy clkaBI"
        >
          Select
        </button>
      </div>
    </div>
    <div
      class="sc-AxiKw jhuQxN"
    >
      <label
        style="padding-right: 15px;"
      >
        Dir: Vertical
      </label>
      <button
        style="padding-right: 15px;"
      >
        Rotate
      </button>
    </div>
  </div>
</div>
`;
